### PR TITLE
I/6098: Add `'px'` default unit to commands operationg on lengths.

### PR DESCRIPTION
**Pull request by jodator**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Table and table cell properties commands add default `'px'` unit to numeric values. Closes ckeditor/ckeditor5#6098.

---

### Additional information

I've added this behavior to the commands:
- `TableWidthCommand`
- `TableHeightCommand`
- `TableCellWidthCommand`
- `TableCellHeightCommand`
- `TableCellPaddingCommand`
- `TableCellBorderWidthCommand`
- `TableBorderWidthCommand`

The logic behind this (`addDefaultUnitToNumericValue()` method) is to add default unit only to numeric values. Other values are passed as-is. This is done because other commands also work like this. Value validation is done in the UI.

### CODE REVIEW

**Comment by jodator** [comment]
The feature is ready to review. The only test that is failing is
> Table cell paragraph post-fixer should wrap in paragraph $text nodes placed directly in tableCell (on table cell modification)

which is the same failing on master so :man_shrugging: I think we can merge it if only this test fails.

File: src/commands/utils.js [hunk 1]
```diff
@@ -85,3 +85,31 @@ export function getSingleValue( objectOrString ) {
 		return top;
 	}
 }
+
+/**
+ * Add a unit to a value if the value is a number or string with a number.
```
Comment (oleq) [inline src/commands/utils.js]:
```suggestion
 * Adds a unit to a value if the value is a number or a string representing a number.
```

File: src/commands/utils.js [hunk 2]
```diff
@@ -85,3 +85,31 @@ export function getSingleValue( objectOrString ) {
...
+ *
+ * Does nothing to a non-numeric values.
```
Comment (oleq) [inline src/commands/utils.js]:
```suggestion
 * **Note**: It does nothing to non-numeric values.
```

File: src/tablecellproperties/commands/tablecellborderwidthcommand.js [hunk 3]
```diff
@@ -22,6 +22,14 @@ import { getSingleValue } from '../../commands/utils';
  *			value: '5px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellborderwidthcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tablecellproperties/commands/tablecellborderwidthcommand.js [hunk 4]
```diff
@@ -22,6 +22,14 @@ import { getSingleValue } from '../../commands/utils';
...
+ *
+ *		editor.execute( 'tableCellBorderWidth', {
+ *			value: '5'
+ *		} );
+ *
+ * Will set `borderWidth` attribute to `'5px'` in the model.
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellborderwidthcommand.js]:
```suggestion
 * will set the `borderWidth` attribute to `'5px'` in the model.
```

File: src/tablecellproperties/commands/tablecellheightcommand.js [hunk 5]
```diff
@@ -21,6 +22,14 @@ import TableCellPropertyCommand from './tablecellpropertycommand';
  *			value: '50px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellheightcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tablecellproperties/commands/tablecellheightcommand.js [hunk 6]
```diff
@@ -21,6 +22,14 @@ import TableCellPropertyCommand from './tablecellpropertycommand';
...
+ *
+ *		editor.execute( 'tableCellHeight', {
+ *			value: '50'
+ *		} );
+ *
+ * Will set `height` attribute to `'50px'` in the model.
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellheightcommand.js]:
```suggestion
 * will set the `height` attribute to `'50px'` in the model.
```

File: src/tablecellproperties/commands/tablecellpaddingcommand.js [hunk 7]
```diff
@@ -22,6 +22,13 @@ import { getSingleValue } from '../../commands/utils';
  *			value: '5px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellpaddingcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tablecellproperties/commands/tablecellpaddingcommand.js [hunk 8]
```diff
@@ -22,6 +22,13 @@ import { getSingleValue } from '../../commands/utils';
...
+ *
+ *		editor.execute( 'tableCellPadding', {
+ *			value: '5'
+ *		} );
+ *
+ * Will set `padding` attribute to `'5px'` in the model.
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellpaddingcommand.js]:
```suggestion
 * will set the `padding` attribute to `'5px'` in the model.
```

File: src/tablecellproperties/commands/tablecellpropertycommand.js [hunk 9]
```diff
@@ -86,4 +87,15 @@ export default class TableCellPropertyCommand extends Command {
 
 		return tableCell.getAttribute( this.attributeName );
 	}
+
+	/**
+	 * Returns proper model value. Can be used to add default unit to numeric values.
```
Comment (oleq) [inline src/tablecellproperties/commands/tablecellpropertycommand.js]:
```suggestion
	 * Returns the proper model value. Can be used to add a default unit to numeric values.
```

File: src/tableproperties/commands/tableborderwidthcommand.js [hunk 10]
```diff
@@ -22,6 +22,14 @@ import { getSingleValue } from '../../commands/utils';
  *			value: '5px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tableproperties/commands/tableborderwidthcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tableproperties/commands/tableborderwidthcommand.js [hunk 11]
```diff
@@ -22,6 +22,14 @@ import { getSingleValue } from '../../commands/utils';
...
+ *
+ *		editor.execute( 'tableBorderWidth', {
+ *			value: '5'
+ *		} );
+ *
+ * Will set `borderWidth` attribute to `'5px'` in the model.
```
Comment (oleq) [inline src/tableproperties/commands/tableborderwidthcommand.js]:
```suggestion
 * Will set the `borderWidth` attribute to `'5px'` in the model.
```

File: src/tableproperties/commands/tableheightcommand.js [hunk 12]
```diff
@@ -21,6 +22,14 @@ import TablePropertyCommand from './tablepropertycommand';
  *			value: '500px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tableproperties/commands/tableheightcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tableproperties/commands/tableheightcommand.js [hunk 13]
```diff
@@ -21,6 +22,14 @@ import TablePropertyCommand from './tablepropertycommand';
...
+ *
+ *		editor.execute( 'tableHeight', {
+ *			value: '50'
+ *		} );
+ *
+ * Will set `height` attribute to `'50px'` in the model.
```
Comment (oleq) [inline src/tableproperties/commands/tableheightcommand.js]:
```suggestion
 * Will set the `height` attribute to `'50px'` in the model.
```

File: src/tableproperties/commands/tablepropertycommand.js [hunk 14]
```diff
@@ -85,4 +86,15 @@ export default class TablePropertyCommand extends Command {
 
 		return table.getAttribute( this.attributeName );
 	}
+
+	/**
+	 * Returns proper model value. Can be used to add default unit to numeric values.
```
Comment (oleq) [inline src/tableproperties/commands/tablepropertycommand.js]:
```suggestion
	 * Returns the proper model value. Can be used to add a default unit to numeric values.
```

File: src/tableproperties/commands/tablewidthcommand.js [hunk 15]
```diff
@@ -21,6 +22,14 @@ import TablePropertyCommand from './tablepropertycommand';
  *			value: '400px'
  *		} );
  *
+ * **Note**: This command adds a default `'px'` unit to a numeric values. Executing:
```
Comment (oleq) [inline src/tableproperties/commands/tablewidthcommand.js]:
```suggestion
 * **Note**: This command adds the default `'px'` unit to numeric values. Executing:
```

File: src/tableproperties/commands/tablewidthcommand.js [hunk 16]
```diff
@@ -21,6 +22,14 @@ import TablePropertyCommand from './tablepropertycommand';
...
+ *
+ *		editor.execute( 'tableWidth', {
+ *			value: '50'
+ *		} );
+ *
+ * Will set `width` attribute to `'50px'` in the model.
```
Comment (oleq) [inline src/tableproperties/commands/tablewidthcommand.js]:
```suggestion
 * Will set the `width` attribute to `'50px'` in the model.
```

**Comment by oleq** [comment]
This is [the place](https://github.com/ckeditor/ckeditor5-table/blob/i/6098/src/ui/utils.js#L164-L168) where the validation should be improved.

**Comment by jodator** [comment]
> Also since the [validation is now working](https://github.com/ckeditor/ckeditor5/issues/6131#event-3022246993), the validators must be adjusted

Bu the validation is still in progress: ckeditor/ckeditor5#6201. It should be ready for tomorrow morning ;)

**Comment by oleq** [comment]
OK then. I somehow missed that.

**Comment by jodator** [comment]
> OK then. I somehow missed that.

But maybe you're right ... because I don't know if `500` will pass that validator and this is UI case.

**Comment by jodator** [comment]
@oleq I'll fix that - that check is UI thing.

**Comment by jodator** [comment]
@oleq I've improved the validation to allow numbers without the unit in the UI. The length validator will be updated in ckeditor/ckeditor5#6201

**Pull request merged**